lib/reverse_sentence.rb

```diff
@@ -1,6 +1,51 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n*m) where n is the length of the sentence and m is the length of the longest word
+# Space complexity: O(1)
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+  return my_sentence if my_sentence.nil?
+
+  reverse(my_sentence)
+
+  i = 0
+  word_length = 0
+
+  while i < my_sentence.length - 1
+    until my_sentence[i] == " " || my_sentence[i].nil?
+      word_length += 1
+      i += 1
+    end
+
+    if word_length > 0
+      index_when_word_starts = i - word_length
+      index_after_word_ends = i
+      my_sentence[index_when_word_starts...index_after_word_ends] = reverse(my_sentence[index_when_word_starts...index_after_word_ends])
```

Comment:
Note that `my_sentence[index_when_word_starts...index_after_word_ends]` creates a new array which is a copy of the original O(n) space complexity.

You can get O(1) space complexity passing the indexes between which you want to reverse instead.